docs/reference/modules/uploadfs.md

```diff
@@ -0,0 +1,359 @@
+
+## Featured methods
+
+### `copyIn(localPath, path, options, callback)`
+This method is used for handling uploads by the various storage types exposed by `uploadfs`. It is used by both the `asset` and `attachment` modules, but can also be used to handle files that aren't covered by the `attachment` module. The `copyIn()` method takes a local filename through the `localPath` argument. This file is copied to a uploadfs path specified by the `path` argument. Any necessary intermediate folders that don't exist will be created. The optional `options` argument is used to pass additional options to the selected storage. Finally, the callback receives any errors. It is often useful to wrap this function in  `promisify` to convert it to a promise-based function.
+
+#### Example
+```javascript
+async handleFile( file, uploadfsPath) {  
+  const copyIn = require('util').promisify(self.uploadfs.copyIn);
+  try {
+    await copyIn(file, uploadfsPath)
+  } catch (e) {
+    console.error(e);
+};  
+```
+
+### `copyOut(localPath, path, options, callback)`
+Much like the `copyIn` method, the `copyOut` method takes a `localPath` and a uploadfs `path` and copies the file back from uploadfs to the local file system. This is used when an existing asset needs to be updated, such as supplying additional cropping or other image processing. It should be used sparingly because it can result in poor storage service performance. Like the `copyIn` method, wrapping `copyOut` in `promisify` is often useful.
+
+#### Example
+
+```javascript
+async modifyImage( file, uploadfsPath) {  
+  const copyOut = require('util').promisify(self.uploadfs.copyOut);
+  const copyIn = require('util').promisify(self.uploadfs.copyIn);
+  try {  
+    await copyOut(file, uploadfsPath);
+    await newImageCrop(file);
+    await copyIn(file, uploadfsPath);
+   } catch (err) {
+    console.error(err);
+  }
+};
+```
+
+### `migrateFromDisabledFileKey(callback)`
+The `migrateFromDisabledFileKey` function provides a way to switch any file previously disabled using the `disabledFileKey` option to use the `disable` method instead. Note: the `disabledFileKey` option should be removed before invoking this method.
+
+#### Example
+
+```javascript
+self.uploadfs.migrateFromDisabledFileKey(function(e) { … });
+```
+
+### `migrateToDisabledFileKey(callback)`
+The `migrateToDisabledFileKey` function provides a way to switch any file previously disabled using the`disable` method to use the `disabledFileKey` option.
+
+#### Example
+
+```javascript
+self.uploadfs.migrateToDisabledFileKey(function(e) { … });
+```
+
+## ENVIRONMENT variables
```

Comment:
I don't think it makes sense for ENVIRONMENT to be in all caps.

Reply:
Changed - just think caps when I think ENV vars - 😄